lib/pbench/server/indexing_tarballs.py

```diff
@@ -524,67 +565,49 @@ def sighup_handler(*args):
                                 raise
                             except Exception:
                                 pass
-                        # Distinguish failure cases, so we can retry the indexing
-                        # easily if possible.  Different `linkerrdest` directories for
-                        # different failures; the rest are going to end up in
-                        # `linkerrdest` for later retry.
-                        controller_path = linksrc_dir.parent
 
+                        # Distinguish failure cases, so we can retry the indexing
+                        # easily if possible.
+                        #
+                        # Only if the indexing was successful do we request the
+                        # next operation (tool indexing). Otherwise we record
+                        # the error in the `server.errors.index` metadata and
+                        # leave the dataset in INDEXING state. (FIX-ME: should
+                        # we expose an "INDEX_FAILED" primary state which would
+                        # "trap" the failed dataset but enable a transition to
+                        # INDEXING to re-try?)
```

Comment:
_[Highlighting this because, generally, we shouldn't be adding `FIX-ME`'s to the code...so this should be removed before merging...once we answer the question.]_

My immediate response is "I don't know" -- I can see advantages to each.  It seems like leaving it in `INDEXING` state is bad, unless we really think that the problem is transient or can otherwise be resolved automatically.  Otherwise, we should probably "set this dataset aside" until a human can deal with it.  Obviously, that could be done without moving it out of `INDEXING` state, but I think using another state (like `INDEX_FAILED`) would be cleaner.

Reply:
I think we want to be able to retry; and that means, if I add an `INDEX_FAILED` state, we'd need to allow transition from `INDEX_FAILED` to `INDEXING`. (Although I'm not currently allowing the re-try transition from `INDEXING` to `INDEXING`, either.)

Right now, this is leaving it in the `-ING` state and recording an error string in metadata for reference. I suppose I should make this re-tryable by adding a transition.

And then I realize that I've removed all use of the `QUARANTINE` state along with the `quarantine` directory and procedure, so I should remove that state, and probably allow re-try transitions for essentially all of the `-ING` states.